File: app/src/main/java/br/com/medeve/Interfaces/IUsuarioDao.java
package br.com.medeve.Interfaces;

import android.app.Activity;

import br.com.medeve.Models.Usuario;

public interface IUsuarioDao {

    int cadastrarUsuairo(Usuario usuario);

    int entrarUsuario(Usuario usuario);

    void recuperarSenhaUsuario(Usuario usuario);

    void persistirUsuario(Class clazz, Activity activity);

    void sair(Activity activity, Class clazz);
}

File: app/src/main/java/br/com/medeve/Interfaces/IUsuarioDao.kt
package br.com.medeve.Interfaces

import android.app.Activity
import androidx.lifecycle.MutableLiveData
import br.com.medeve.Models.Usuario

interface IUsuarioDao {
    fun cadastrarUsuairo(usuario: Usuario?)
    fun entrarUsuario(usuario: Usuario?)
    fun recuperarSenhaUsuario(usuario: Usuario?)
    fun persistirUsuario(clazz: Class<*>?, activity: Activity?)
    fun sair(activity: Activity?, clazz: Class<*>?)
}